File: 3753-maximum-difference-between-even-and-odd-frequency-i/maximum-difference-between-even-and-odd-frequency-i.py
class Solution:
    def maxDifference(self, s: str) -> int:
        freq = Counter(s)
        even = float('inf')
        odd = 0
        for i in freq.values():
            if i%2 == 0 and i!=0:
                even = min(even,i)
            else:
                odd = max(odd,i)

        return odd - even
        

File: 3753-maximum-difference-between-even-and-odd-frequency-i/maximum-difference-between-even-and-odd-frequency-i.java
class Solution {
    public int maxDifference(String s) {
        int[] freq = new int[26];
        int odd = Integer.MIN_VALUE;
        int even = Integer.MAX_VALUE;
        for(char c : s.toCharArray()){
            freq[c - 'a']++;
        }
        for(int n : freq){
            if(n%2 == 0 && n!=0){
                even = Math.min(even,n);
            }else{
                odd = Math.max(odd,n);
            }
        }

        return odd - even;
    }
}